client/components/survey-container/components/question-step-mapping.tsx

```diff
@@ -0,0 +1,58 @@
+import { QuestionType } from '../types';
+import { QuestionSelectionComponentProps } from './question-step';
+import SurveyCheckboxControl from './survey-checkbox-control';
+import SurveyFlowCardControl from './survey-flow-card';
+import SurveyRadioControl from './survey-radio-control';
+
+type questionTypeComponentMapProps = {
+	[ QuestionType.SINGLE_CHOICE ]: ( {
+		onChange,
+		question,
+		value,
+		disabled,
+		onContinue,
+	}: QuestionSelectionComponentProps ) => JSX.Element;
+	[ QuestionType.MULTIPLE_CHOICE ]: ( {
+		onChange,
+		question,
+		value,
+		disabled,
+		onContinue,
+	}: QuestionSelectionComponentProps ) => JSX.Element;
+};
+
+type questionHeaderButtonsMapProps = {
+	[ key: string ]: {
+		shouldHideContinueButton: boolean;
+		shouldHideSkipButton: boolean;
+	};
+};
+
+export const questionTypeComponentMap = {
+	[ QuestionType.SINGLE_CHOICE ]: SurveyRadioControl,
+	[ QuestionType.MULTIPLE_CHOICE ]: SurveyCheckboxControl,
+};
+
+export const flowQuestionMapping: {
+	[ key: string ]: {
+		questionTypeComponentMap: questionTypeComponentMapProps;
+		headerButtonsMapping: questionHeaderButtonsMapProps;
+	};
+} = {
+	'/start/guided/initial-intent': {
+		questionTypeComponentMap: {
+			[ QuestionType.SINGLE_CHOICE ]: SurveyFlowCardControl,
+			[ QuestionType.MULTIPLE_CHOICE ]: SurveyCheckboxControl,
+		},
+		headerButtonsMapping: {
+			'what-brings-you-to-wordpress': {
+				shouldHideContinueButton: true,
+				shouldHideSkipButton: true,
+			},
+			'what-are-your-goals': {
+				shouldHideContinueButton: false,
+				shouldHideSkipButton: false,
```

Comment:
The reason for hiding the CTAs is to cover required questions? If that is the case, we could consider adding support for required questions to the survey structure so the component could check if there are answers or not, and disable/enable the Continue and Skip buttons.

Reply:
This was initially added to follow the design and the first question was also required. Adding support to hide the CTA and buttons would be great.

Reply:
> we could consider adding support for required questions

We don't need this anynmore, as both questions are now optional, but it would be great to have in place @phcp